UserSync/Controllers/SyncController.cs

```diff
@@ -49,13 +49,12 @@ public class SyncController : ApiController
         [Authorize]
         public async Task GetAsync(string tenantId)
         {
-            MSALCache appTokenCache = new MSALCache(Startup.clientId);
+            MSALMemoryTokenCache tokenCacheHelper = new MSALMemoryTokenCache(Startup.clientId);
 
-            // Get a token for the Microsoft Graph. If this line throws an exception for
-            // any reason, we'll just let the exception be returned as a 500 response
+            // Get a token for the Microsoft Graph. If this line throws an exception for any reason, we'll just let the exception be returned as a 500 response
             // to the caller, and show a generic error message to the user.
             ConfidentialClientApplication daemonClient = new ConfidentialClientApplication(Startup.clientId, string.Format(AuthorityFormat, tenantId), Startup.redirectUri,
-                                                                                           new ClientCredential(Startup.clientSecret), null, appTokenCache.GetMsalCacheInstance());
+                                                                                           new ClientCredential(Startup.clientSecret), tokenCacheHelper.GetMsalUserTokenCacheInstance(), tokenCacheHelper.GetMsalAppTokenCacheInstance());
```

Comment:
This seems wrong, @kalyankrishna1 
The user cache should take the user unique id, otherwise the cache will contain several users, which we don't want. In confidential client applications there should be one user token cache per identity signing-in.

Reply:
fixed